src/parse/ast/bound.ts

```diff
@@ -27,12 +67,10 @@ export const writeDoubleRightBoundMessage = <root extends string>(
 type writeDoubleRightBoundMessage<root extends string> =
     `Expression ${root} must have at most one right bound`
 
-type isBoundable<data> = [data] extends [SizedData] ? true : false
-
 export const writeUnboundableMessage = <root extends string>(
     root: root
 ): writeUnboundableMessage<root> =>
-    `Bounded expression ${root} must be a number, string, Array, or Date`
+    `Bounded expression ${root} must be a number, string or Array`
 
 type writeUnboundableMessage<root extends string> =
```

Comment:
Is this only used for numeric bounds? If so we should update the name to `writeNumericallyUnboundableMessage`